Snowflake.Data.Tests/IntegrationTests/SFBindTestIT.cs

```diff
@@ -769,5 +774,290 @@ public void testPutArrayBind1()
+        private readonly DateTimeOffset? _expectedDateTimeOffset;
+
+        internal static ExpectedTimestampValueProvider From(string timestampWithTimeZone, SFDataType columnType)
+        {
+            if (IsOffsetType(columnType))
+                return new ExpectedTimestampValueProvider(DateTimeOffset.ParseExact(timestampWithTimeZone, 
+                        "yyyy/MM/dd HH:mm:ss.fff zzz", 
+                        CultureInfo.InvariantCulture), columnType); 
+                    
+            return new ExpectedTimestampValueProvider(DateTime.ParseExact(timestampWithTimeZone, 
+                    "yyyy/MM/dd HH:mm:ss.fff zzz", 
+                    CultureInfo.InvariantCulture), columnType);
+        }
+
+        private ExpectedTimestampValueProvider(DateTime dateTime, SFDataType columnType)
+        {
+            _expectedDateTime = dateTime;
+            _expectedDateTimeOffset = null;
+            _columnType = columnType;
+        }
+
+        private ExpectedTimestampValueProvider(DateTimeOffset dateTimeOffset, SFDataType columnType)
+        {
+            _expectedDateTimeOffset = dateTimeOffset;
+            _expectedDateTime = null;
+            _columnType = columnType;
+        }
+
+        internal SFDataType ExpectedColumnType() => _columnType;
+
+        internal void IsEqual(object actual, string comparisonFormat, string faultMessage)
+        {
+            switch (_columnType)
+            {
+                case SFDataType.TIMESTAMP_TZ:
+                    Assert.AreEqual(GetDateTimeOffset().ToString(comparisonFormat),
+                        ((DateTimeOffset)actual).ToString(comparisonFormat),
+                        faultMessage);
+                    break;
+                case SFDataType.TIMESTAMP_LTZ:
+                    Assert.AreEqual(GetDateTimeOffset().ToUniversalTime().ToString(comparisonFormat),
+                        ((DateTimeOffset)actual).ToUniversalTime().ToString(comparisonFormat),
+                        faultMessage);
+                    break;
+                default:
+                    Assert.AreEqual(GetDateTime().ToString(comparisonFormat),
+                        ((DateTime)actual).ToString(comparisonFormat),
+                        faultMessage);
+                    break;
+            }
+        }
+
+        internal DateTime GetDateTime() => _expectedDateTime ?? throw new Exception($"Column {_columnType} is not matching the expected value type {typeof(DateTime)}");
+
+        internal DateTimeOffset GetDateTimeOffset() => _expectedDateTimeOffset ?? throw new Exception($"Column {_columnType} is not matching the expected value type {typeof(DateTime)}");
+        
+        private static bool IsOffsetType(SFDataType type) => type == SFDataType.TIMESTAMP_LTZ || type == SFDataType.TIMESTAMP_TZ;
+    }
+    
+    public enum SFTableType 
+    {
+        Standard,
+        Hybrid,
+        Iceberg
+    }
+    
+    static class TestExtensions
```

Comment:
this is not used, the tests are using SFBaseTest. I'm writing a comment just to let you think if we should add it in this PR since there is no usage of this code. Another question would be why in this 1000 lines long SFBindTestID?

Reply:
Code doesn't compile without this code. It's an integral part of this test.
I would maybe even consider in the future those Add, CreateCommand, ExecuteNonQuery methods become a production interface. For now if needed we can move it outside this test class although some of it work along just with the ExpectedTimeoutValueProvider class. So why to move it away from here?

Reply:
For now moved it so slim this class. Have a look @sfc-gh-knozderko